### PR TITLE
Add systemd steps to Grafana Agent standalone binary docs

### PR DESCRIPTION
**Pull request by clayton-cornell**
#### PR Description

This PR adds systemd steps to the Grafana Agent Flow mode documentation. This information was removed from other areas of the documentation. Since its removal, there have been multiple requests to add it back in. This is also relevant information for some of the prerequisites to setting up Grafana Agent in configuration management tooling.

#### PR Checklist

- [ ] CHANGELOG.md updated
- [ ] Documentation added
- [ ] Tests updated
- [ ] Config converters updated

### CODE REVIEW

File: docs/sources/flow/setup/start-agent.md [hunk 1]
```diff
@@ -172,4 +172,40 @@ Replace the following:
...
+
+1. Create a file in `/etc/systemd/system` called `grafana-agent.service` with the following contents.
+
+   ```shell
+   [Unit]
+   Description=Grafana Agent
+
+   [Service]
+   User=grafana-agent
+   ExecStart=BINARY_PATH/grafana-agent --config.file=/etc/grafana-agent-flow.river
```
Comment (rfratto) [inline docs/sources/flow/setup/start-agent.md:192]: That's the wrong CLI for Flow: 

```suggestion
   ExecStart=BINARY_PATH/grafana-agent run /etc/grafana-agent-flow.river
```

Comment (clayton-cornell) [reply docs/sources/flow/setup/start-agent.md:193]: D'oh, I missed that.

File: docs/sources/flow/setup/start-agent.md [hunk 2]
```diff
@@ -172,4 +172,40 @@ Replace the following:
...
+   Restart=always
```
Comment (rfratto) [inline docs/sources/flow/setup/start-agent.md:194]: There's a few more settings we provide in our service files: 

https://github.com/grafana/agent/blob/main/packaging/grafana-agent-flow/deb/grafana-agent-flow.service

I think we should at _least_ bring over ExecReload so `systemctl reload grafana-agent` works.

File: docs/sources/flow/setup/start-agent.md [hunk 3]
```diff
@@ -172,4 +172,41 @@ Replace the following:
...
+
+You can set up and manage the standalone binary for Grafana Agent as a Linux systemd service.
+
+1. To create a new user called `grafana-agent` run the following command in a terminal window:
+
+   ```shell
+   sudo useradd --no-create-home --shell /bin/false grafana-agent
+   ```
+
+1. Create a file in `/etc/systemd/system` called `grafana-agent.service` with the following contents:
```
Comment (rfratto) [inline docs/sources/flow/setup/start-agent.md:185]: Should this be called `grafana-agent-flow.service` to align with what the system packages install?

**Review by rfratto — APPROVED** [review]
LGTM, though I'm a little nervous about this one; we provide opinionated service files with our system packages, so it feels like this one could open the door to people introducing fields that break things and opening issues.

Should this be labeled as "use at your own risk," or some similar "please know what you're doing" warning?

File: docs/sources/flow/setup/start-agent.md (hunk 2 above)
Comment (clayton-cornell) [reply docs/sources/flow/setup/start-agent.md:194]: Updated the service file contents

**Comment by clayton-cornell** [comment]
@rfratto Do you want one more look at this before merging?

**Review by rfratto — APPROVED** [review]
LGTM, thanks Clayton!

File: docs/sources/flow/setup/start-agent.md [hunk 4]
```diff
@@ -150,26 +150,102 @@ If you downloaded the standalone binary, you must run the agent from a terminal
...
 ```shell
-AGENT_MODE=flow BINARY_PATH run CONFIG_FILE
+AGENT_MODE=flow BINARY_PATH run CONFIG_PATH
 ```
 
 Replace the following:
 
-* `BINARY_PATH`: The path to the Grafana Agent binary file
-* `CONFIG_FILE`: The path to the Grafana Agent configuration file.
+* `BINARY_PATH`: The path and Grafana Agent binary filename.
```
Comment (jdbaldry) [inline docs/sources/flow/setup/start-agent.md:158]: I'm not sure that the new language makes sense.

A path includes the filename already.

File: docs/sources/flow/setup/start-agent.md [hunk 5]
```diff
@@ -150,26 +150,102 @@ If you downloaded the standalone binary, you must run the agent from a terminal
...
+
+1. To create a new user called `grafana-agent-flow` run the following command in a terminal window:
+
+   ```shell
+   sudo useradd --no-create-home --shell /bin/false grafana-agent-flow
+   ```
+
+1. Create a service file in `/etc/systemd/system` called `grafana-agent-flow.service` with the following contents:
+
+   ```shell
```
Comment (jdbaldry) [inline docs/sources/flow/setup/start-agent.md:191]: This is not a shell snippet.

PrismJS supposedly supports the info string `systemd` for unit configuration file syntax highlighting.

File: docs/sources/flow/setup/start-agent.md (hunk 4 above)
Comment (clayton-cornell) [reply docs/sources/flow/setup/start-agent.md:158]: Updated in https://github.com/grafana/agent/pull/5455

File: docs/sources/flow/setup/start-agent.md (hunk 5 above)
Comment (clayton-cornell) [reply docs/sources/flow/setup/start-agent.md:191]: Updated in https://github.com/grafana/agent/pull/5455

**Pull request merged**